docs/aws/audit/codebuildmonitoring/rules/codebuild_project_artifact_encryption_remediation.mdx

```diff
@@ -1,7 +1,276 @@
+3. **Create/Update Project**: Use the `create_project` or `update_project` method to configure artifact encryption.
+4. **Artifact Configuration**: Set `encryptionDisabled` to `False` to enable encryption for the artifacts.
 
+This script checks if the CodeBuild project exists and either updates it or creates a new one with artifact encryption enabled. Adjust the parameters such as `project_name`, `bucket_name`, and other configurations as needed.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the CodeBuild service.
+2. In the navigation pane, select "Build projects". This will display a list of all the CodeBuild projects in your AWS account.
+3. Select the CodeBuild project that you want to check for Artifact Encryption. This will open the project details page.
+4. In the project details page, scroll down to the "Artifacts" section. Check if the "Encryption key" field is set to a specific AWS Key Management Service (KMS) key. If this field is set to "AWS Managed Key" or a specific KMS key, then Artifact Encryption is enabled for the CodeBuild project. If this field is not set or is set to "Disabled", then Artifact Encryption is not enabled for the CodeBuild project.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the CodeBuild projects in your account using the following command:
+
+   ```
+   aws codebuild list-projects
+   ```
+
+   This command will return a list of all the CodeBuild project names in your account.
+
+3. Next, you can describe a specific CodeBuild project to get its details. Replace 'project-name' with the name of your project:
+
+   ```
+   aws codebuild batch-get-projects --names project-name
+   ```
+
+   This command will return a JSON object that contains the details of the specified CodeBuild project.
+
+4. Finally, you can check if artifact encryption is enabled for the project by looking at the 'encryptionKey' field in the returned JSON object. If the 'encryptionKey' field is set to 'aws/s3', it means that the default S3 key is being used, which is not recommended. If the 'encryptionKey' field is set to a specific KMS key, it means that artifact encryption is enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Artifact Encryption is enabled for CodeBuild Project in AWS, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Set up AWS SDK for Python (Boto3):**
+   First, you need to set up Boto3. Install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>